pkgs/by-name/pr/proton-vpn-local-agent/package.nix

```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  python3,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "proton-vpn-local-agent";
+  version = "unstable-20241010";
+  cargoHash = "sha256-42cOoBE9ebqSiENOTTzGV5zvxwY+qm3+zvsT1uGZZWg=";
+
+  src = fetchFromGitHub {
+    owner = "ProtonVPN";
+    repo = "python-proton-vpn-local-agent";
+    rev = "01332194d217d91a514ecaebcdfbfa3d21ccd1ed";
+    hash = "sha256-I+tbVQzD4xJUsoRF8TU/2EMldVqtfxY3E7PQN3ks0mA=";
+  };
+
+  sourceRoot = "${src.name}/python-proton-vpn-local-agent";
+
+  installPhase = ''
+    # manually install the python binding
+    mkdir -p $out/${python3.sitePackages}/proton/vpn/
+    cp ./target/x86_64-unknown-linux-gnu/release/libpython_proton_vpn_local_agent.so $out/${python3.sitePackages}/proton/vpn/local_agent.so
```

Comment:
`${stdenv.hostPlatform.rust.cargoShortTarget}`

Reply:
> Thanks for this PR. This line is hardcoded to a specific arch, does the local agent only work on x86_64-linux, or can it be run on others? You might also want to put the platforms that are supported in the meta field so that nix will alert if building on a non-supported arch

For now I've marked the supported architectures with 'lib.platforms.linux' same as the protonvpn app.